src/zui/ZUITextfieldToClipboard.tsx

```diff
@@ -0,0 +1,41 @@
+import copy from 'copy-to-clipboard';
+import { FormattedMessage as Msg } from 'react-intl';
+import { Box, Button } from '@mui/material';
+import React, { useEffect, useState } from 'react';
+
+const ZUITextfieldToClipboard: React.FunctionComponent<{
+  children: React.ReactNode;
+  copyText: string | number | boolean;
+}> = ({ children, copyText }) => {
+  const [copied, setCopied] = useState<boolean>(false);
+
+  const handleClick = () => {
+    copy(copyText.toString());
+    setCopied(true);
+    setTimeout(() => setCopied(false), 2500);
+  };
+
+  useEffect(() => {
+    if (copied) {
+      setCopied(false);
+    }
+  }, []);
```

Comment:
Perhaps not that necessary... I could remove its since the setTimeout resets `copied`s value

Reply:
Does this even do anything? When the dependency array is empty, the effect will run when the component is mounted. When a component is mounted, it's state is already at it's default state, i.e. `false` in this case.

I think you did this to handle situations where the user navigates away and comes back, to make sure that `copied` is false when they return. But in that case, the component will have unmounted (when navigating away) and when it remounts, the default state is set on line 10.

Reply:
You are right! Thought about it for a bit and then tested just to be sure